lib/player.js

```diff
@@ -1694,17 +1694,24 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     // initialize it now even through it appears a little out-of-place.
     this.regionTimeline_ =
         new shaka.media.RegionTimeline(() => this.seekRange());
-    this.regionTimeline_.setListeners(/* onRegionAdded= */ (region) => {
+    this.regionTimeline_.addEventListener('regionadd', (event) => {
+      /** @type {shaka.extern.TimelineRegionInfo} */
+      const region = event['region'];
       this.onRegionEvent_(shaka.Player.EventName.TimelineRegionAdded, region);
+
       if (this.adManager_) {
         this.adManager_.onDashTimedMetadata(region);
       }
     });
+
     this.qualityObserver_ = null;
     if (this.config_.streaming.observeQualityChanges) {
       this.qualityObserver_ = new shaka.media.QualityObserver(
           () => this.getBufferedInfo());
-      this.qualityObserver_.setListeners((mediaQualityInfo, position) => {
+
+      this.qualityObserver_.addEventListener('qualitychange', (event) => {
+        const mediaQualityInfo = event['quality'];
+        const position = event['position'];
```

Comment:
Is the compiler automatically able to infer the types here? If not, please add explicit type casting for these fields like the other event listeners (e.g., L1698).

Reply:
Good call.  It isn't able to infer.  Done!